cmake/std/pr_python_test_driver.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+#
+# Prepare Environment
+#
+
+# The existence of PYTHONHOME can cause some odd problems
+# if we're seeing errors that look like `No module named 'encodings'`
+# then a usual fix is to try and unset PYTHONHOME.
+# I think this is due to a conflict in modules loaded by SEMS and then
+# our use of a specific Python that's different.
+# - See: https://stackoverflow.com/q/38132755/2059999 for more information
+unset PYTHONHOME
+
+
+#
+# Get pip
+# - @param1 python_exe - the python executable to install PIP for
+function get_pip() {
+    local python_exe=${1:?}
+
+    echo -e "--- Python: ${python_exe:?}"
+
+    # fetch get-pip.py
+    echo -e "--- curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py"
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+
+    get_pip_args=(
+        --user
+        --proxy="http://wwwproxy.sandia.gov:80"
+        --no-setuptools
+        --no-wheel
+    )
+    echo -e ""
+    echo -e "--- ${python_exe:?} ./get-pip.py ${get_pip_args[@]}"
+    ${python_exe:?} ./get-pip.py ${get_pip_args[@]}
+}
+
+
+#
+# Install Python pacakges using pip
+#
+# - @param1 pip_exe - the pip binary to use, i.e., pip3.
+#
+function get_python_packages() {
+    local pip_exe=${1:?}
+
+    echo -e "--- Pip   : ${pip_exe:?}"
+
+    pip_args=(
+        --use-feature=2020-resolver
+        configparser
+        mock
+        pytest
+        pytest-cov
+    )
+    echo -e "--- ${pip_exe:?} install --user ${pip_args[@]}"
+    ${pip_exe:?} install --user ${pip_args[@]}
+}
```

Comment:
I think I already saw these two functions elsewhere.  Should they be pulled out into some other file that's sourced instead?

Reply:
Yeah, I think it's cut and pasted into two places.

Generally I do have a `common.bash` script where I keep a single copy of functions like these... I'm not heavily optimizing the bash scripts in this PR... most of the legacy bash in Trilinos doesn't really use functions, it's generally just all in-line.  

I think a cleanup pass on stuff like this is worth doing in a follow-on PR.